apps/src/code-studio/components/Libraries/LibraryCreationDialog.jsx

```diff
@@ -67,33 +90,85 @@ class LibraryCreationDialog extends React.Component {
     );
   };
 
+  validateInput = () => {
+    // Check if any of the checkboxes are checked
+    // If this changes the publishable state, update
+    let isChecked = $("input[type='checkbox']").is(':checked');
+    if (isChecked !== this.state.canPublish) {
+      this.setState({canPublish: isChecked});
+    }
+  };
+
   displayFunctions = () => {
     if (!this.state.loadingFinished) {
       return <div>Loading...</div>;
     }
+    let keyIndex = 0;
     return (
       <div>
-        <div>{this.state.libraryName}</div>
-        {this.state.selectedFunctionList.map(selectedFunction => {
-          let name = selectedFunction.functionName;
-          return <div key={name}>{name}</div>;
-        })}
+        <Heading2>
+          <b>Library Name: </b>
+          {this.state.libraryName}
+        </Heading2>
+        <form onSubmit={this.publish}>
+          <textarea
+            required
+            name="description"
+            rows="2"
+            cols="200"
+            style={styles.textarea}
+            placeholder="Write a description of your library"
+          />
+          {this.state.selectedFunctionList.map(selectedFunction => {
+            let name = selectedFunction.functionName;
+            let comment = selectedFunction.comment;
+            return (
+              <div key={keyIndex++} style={styles.functionItem}>
+                <input
+                  type="checkbox"
+                  style={styles.largerCheckbox}
+                  disabled={comment.length === 0}
+                  onClick={this.validateInput}
+                />
+                {name}
+                <br />
+                {comment.length === 0 && (
+                  <p style={styles.alert}>
+                    This function cannot be exported until you add a comment to
+                    it.
+                  </p>
+                )}
+                <pre>{comment}</pre>
+              </div>
+            );
+          })}
+          <input
+            className="btn btn-primary"
+            type="submit"
+            value={i18n.publish()}
+            disabled={!this.state.canPublish}
+          />
+        </form>
       </div>
     );
   };
 
   render() {
     return (
-      <BaseDialog
+      <Dialog
         isOpen={this.props.dialogIsOpen}
         handleClose={this.handleClose}
         useUpdatedStyles
       >
-        {this.displayFunctions()}
-        <button type="button" onClick={this.publish}>
-          {i18n.publish()}
-        </button>
-      </BaseDialog>
+        <Body>
+          <PadAndCenter>
+            <div style={styles.libraryBoundary}>
+              <Heading1>Functions as a Library</Heading1>
```

Comment:
Maybe "Export functions as a library?" (also posted in #student-created-libraries where more of the UI review is happening)

Reply:
:heavy_check_mark: